### PR TITLE
[INLONG-10152][Sort] Refactor MetricOption code structure.

### LINKED ISSUE #10152: [Improve][Sort] Refactor MetricOption code structure.
### Description

The MetricOption constructor is not concise. checkArgument method in build() will be better.

https://github.com/apache/inlong/blob/bf42e1a5f8f204c4111d012cbd68256f8f870449/inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java#L59-L108

https://github.com/apache/inlong/blob/3b70a93d96b74bf8250d8f08da43e0f5a6acbf56/inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java#L240-L246

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
Prepare a Pull Request

- [INLONG-10152][Sort] Refactor MetricOption code structure.

- Fixes #10152 

### Motivation

The MetricOption constructor Does not conform to the single responsibility design pattern，check arguments not exist in constructor will make code clearly.

### Modifications

Adjust constructor and move check arguments code statement.

### CODE REVIEW

File: inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java [hunk 1]
```diff
@@ -57,54 +57,26 @@ public class MetricOption implements Serializable {
...
             long initBytes,
             Long initDirtyRecords,
             Long initDirtyBytes,
             Long readPhase,
-            String inlongAuditKeys) {
-        Preconditions.checkArgument(!StringUtils.isNullOrWhitespaceOnly(inlongLabels),
-                "Inlong labels must be set for register metric.");
-
+            List<Integer> inlongAuditKeys,
+            HashSet<String> ipPortList) {
```
Comment (vernedeng) [inline inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:69]: Use Set instead of HashSet

Comment (vernedeng) [reply inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:69]: It's called list, but a set

Comment (XiaoYou201) [reply inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java]: Renamed and refactor the hashSet

**Review by aloyszhang — APPROVED** [review]
LGTM

File: inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java [hunk 2]
```diff
@@ -57,62 +58,38 @@ public class MetricOption implements Serializable {
...
+        this.registeredMetric = registeredMetric;
     }
 
     public Map<String, String> getLabels() {
         return labels;
     }
 
-    public HashSet<String> getIpPortList() {
-        return ipPortList;
+    public Set<String> getIpPortSet() {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:87]: method is never used

File: inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java [hunk 3]
```diff
@@ -57,62 +58,38 @@ public class MetricOption implements Serializable {
...
+        return ipPortSet;
+    }
+
+    public HashSet<String> getIpPortSetAsHashSet() {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:91]: getIpPorts() is enough for method name

File: inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java (hunk 2 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:87]: This method was originally provided for the Audit module, but the Audit module uses HashSet, and this method returns Set, so a getIpPortSetAsHashSet method was added to be compatible with Audit.
企业微信截图_4bb0f4fd-016f-4228-9231-f8a1cffa1825

File: inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java (hunk 3 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/metric/MetricOption.java:91]: This method was originally provided for the Audit module, but the Audit module uses HashSet, and this method returns Set, so a getIpPortSetAsHashSet method was added to be compatible with Audit.
企业微信截图_4bb0f4fd-016f-4228-9231-f8a1cffa1825

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**